src/components/ContractTable.tsx

```diff
@@ -0,0 +1,178 @@
+import { LoadingOutlined, SyncOutlined } from "@ant-design/icons";
+import { Button, Input, Table } from "antd";
+import fuzzysort from "fuzzysort";
+import intersectionBy from "lodash/intersectionBy";
+import React, { useCallback, useEffect, useState } from "react";
+import { useRecords } from "../hooks/useRecords";
+import SDKSession from "../sdk/sdkSession";
+// import { Record } from "../types/records";
+import { constants } from "../util/helpers";
+const { CONTRACTS_PER_PAGE } = constants;
+
+/**
+ * `ContractTable` is a table of ABI contract data with some management features to
+ * make it easier to manage a large amount of contracts.
+ *
+ * @param `session` - the active SDK session
+ */
+export const ContractTable = ({ session }: { session: SDKSession }) => {
+  const [input, setInput] = useState("");
+  const [isLoading, setIsLoading] = useState(false);
+  const [contracts, addContracts, removeContracts] = useRecords([]);
+  const [filteredContracts, setFilteredContracts] = useState([]);
+  const [selectedContracts, setSelectedContracts] = useState([]);
+
+  const fetchRecords = useCallback(
+    async (fetched = 0, retries = 1) => {
+      setIsLoading(true);
+      const res: any = await session.client
+        .getAbiRecords({
+          n: 10,
+          startIdx: fetched,
+          category: "",
+        })
+        .catch((err) => {
+          setIsLoading(false);
+          return console.error(err);
+        });
+
+      const _contracts = res.records.map((r) => ({ id: r.header.name, ...r }));
+      addContracts(_contracts);
+      const totalFetched = res.numFetched + fetched;
+      const remainingToFetch = res.numRemaining;
+      if (remainingToFetch > 0) {
+        fetchRecords(totalFetched);
+      } else {
+        setIsLoading(false);
+      }
+    },
+    [session, addContracts]
+  );
+
+  useEffect(() => {
+    fetchRecords();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    setInput("");
+    setFilteredContracts(contracts);
+  }, [contracts, isLoading]);
+
+  const filter = useCallback(
+    (value) =>
+      fuzzysort
+        .go(value, contracts, { keys: ["header.name", "header.sig"] })
+        .map((x) => x.obj),
+    [contracts]
+  );
+
+  const handleOnSelect = (_, __, _selectedContracts) => {
+    setSelectedContracts(_selectedContracts);
+  };
+
+  const handleOnSelectAll = (_, _selectedContracts) => {
+    setSelectedContracts(_selectedContracts);
+  };
+
+  const removeSelected = () => {
+    setIsLoading(true);
+    const sigs = selectedContracts.map((contract) => contract.header.sig);
+    session.client.removeAbiRecords({ sigs }, (err, val) => {
```

Comment:
Feels like SDK calls should not be in `.tsx` files... can you put this somewhere else?

Reply:
Removing all the SDK stuff from components would be a pretty big change. 

I have a branch where I pull out all of the session code and put it into a `useContext` that is made available to components with a custom hook. Do you want me to do it separately in that PR?

Reply:
Sorry thought I deleted this comment. Disregard.